application/forms/IcingaDependencyForm.php

```diff
@@ -212,18 +213,46 @@ protected function addObjectsElement()
 
         if (!empty($sentParent) || $dependency->isApplyRule()) {
             $parentService = $dependency->get('parent_service');
+            if ($parentService === null) {
+                $parentServiceVar = $dependency->get('parent_service_by_name');
+                if ($parentServiceVar) {
+                    $parentService = '$' . $parentServiceVar . '$';
+                }
+            }
+
+            $applyTo = $this->getSentOrObjectValue('apply_to');
+
+            $description = 'Optional. The parent service. If omitted this dependency'
+                . ' object is treated as host dependency.';
+
+            if ($applyTo === 'service') {
+                $description .= ' You might want to refer to Service Custom Variables via $service.vars.varname$';
+            }
+
+            $callbackValidator = new Zend_Validate_Callback(function ($value) {
+                $applyTo = $this->getSentValue('apply_to');
+                if ($applyTo === 'service') {
+                    return true;
+                }
+
+                return ! $this->isCustomVar($value);
+            });
+
+            $callbackValidator->setMessage(
+                $this->translate('The parent service cannot be a custom variable for host dependency'),
+                Zend_Validate_Callback::INVALID_VALUE
+            );
+
             $this->addElement('text', 'parent_service', [
-                    'label' => $this->translate('Parent Service'),
-                    'description' => $this->translate(
-                        'Optional. The parent service. If omitted this dependency'
-                         . ' object is treated as host dependency.'
-                    ),
-                    'class' => "autosubmit director-suggest",
-                    'data-suggestion-context' => 'servicenames',
-                    'data-suggestion-for-host' => $sentParent,
-                    'order' => 20,
-                    'value' => $parentService
-                ]);
+                'label'                    => $this->translate('Parent Service'),
+                'description'              => $this->translate($description),
```

Comment:
As discussed, this doesn't work. You can only translate literal strings, so either the two parts above individually or both variants on their own in full.